.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+# Tests are also run at: https://ci.appveyor.com/project/RodVagg/nan
+# https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: ci
+on:
+  push:
+  #  branches: [main]
+  pull_request:
+    branches: [main]
+jobs:
+  ci:
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [18.x, 20.x, 22.x, 23.x]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      # - run: npm ci
+      - run: npm install
+      # - run: npm run build --if-present
+      - run: npm run rebuild-tests-2017
+      # - run: npm test
```

Comment:
Why the commented out lines? I think `npm ci` would be better to use than `npm install`. I also don't see a `rebuild-tests-2017` script in this repo, only `rebuild-tests-2015`.

Reply:
I recently redid the scripts 848a433c5ed6d6540af198c8251ddf45d2174c93